allennlp/models/model.py

```diff
@@ -51,6 +52,27 @@ def __init__(self,
         self.vocab = vocab
         self._regularizer = regularizer
 
+    def cuda_device(self) -> int:
+        """
+        This method checks the device of the first parameter to determine the cuda_device
+        this model is to be run on.  If there are no parameters, it returns None.
+        Returns
+        -------
+        The cuda device this model is to be run on.
+        """
+        device = None
+        for parameter in self.parameters():
```

Comment:
it might be a little clearer/pythonic to check a set of devices, and raise an error if the size of the set is > 1:

```
devices = {util.get_device_of(param) for param in self.parameters()}

if len(devices) != 1:
    raise ConfigurationError ...

else:
    # Only has one element, pop it off and return it.
    return devices.pop()
```

But how you have it is fine.

Reply:
Also, it's possible but unlikely to have a model with no parameters - by default, this function should return -1, I think.

Reply:
It's potentially a slower solution, but it's significantly cleaner so I'll make the change ;-)

Reply:
Interested - why would it be slower?